code/modules/projectiles/guns/ballistic/smg.dm

```diff
@@ -403,3 +392,20 @@ EMPTY_GUN_HELPER(automatic/smg/cm5)
 
 	unfolded_slowdown = 0.35
 	unfolded_wield_delay = 0.4 SECONDS
+
+
+/obj/item/gun/ballistic/automatic/smg/skm_carbine/inteq/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.13 SECONDS)
```

Comment:
Same situation as the tec9.
Use the factored firing mode handling, Sorry I meant for both of them

Reply:
Removed the automatic fire component and it's inheriting the needed behavior. Shooould be good if tests don't explode.